src/metabase/driver/sql/query_processor.clj

```diff
@@ -153,6 +153,12 @@
 (defmethod date [:sql :year-of-era]      [_driver _ expr] (hx/year expr))
 (defmethod date [:sql :week-of-year-iso] [_driver _ expr] (hx/week expr))
 
+(defmulti datetime-diff
+  "Return a HoneySQL form for calculating the datetime-diff for a given unit."
```

Comment:
It's also nice to mention what the method is _for_ -- this method is used by the default implementation of `sql.qp/->honeysql` for `:datetime-diff`, or whatever -- and when or why people would want to implement it (implement it if your driver supports `:datetime-diff` and you want to use the default implementation of `->honeysql` for `:datetime-diff` clauses, which is recommended because ______ (built-in error-checking, etc.))

Reply:
I'm really not trying to be incredibly fussy here, but we have third-party driver authors and we should really error on the side of documenting our driver interface too much

Reply:
Updated https://github.com/metabase/metabase/pull/27237/commits/06eb4b6d880e0a310820f9faa6fdd272d37ea560